src/main/host/syscall/handler/timerfd.rs

```diff
@@ -0,0 +1,244 @@
+use std::sync::Arc;
+
+use atomic_refcell::AtomicRefCell;
+use nix::errno::Errno;
+use nix::sys::timerfd::{TimerFlags, TimerSetTimeFlags};
+use shadow_shim_helper_rs::{
+    emulated_time::EmulatedTime, simulation_time::SimulationTime, syscall_types::ForeignPtr,
+};
+use syscall_logger::log_syscall;
+
+use crate::core::worker::Worker;
+use crate::host::descriptor::{
+    timerfd::TimerFd, CompatFile, Descriptor, DescriptorFlags, File, FileStatus, OpenFile,
+};
+use crate::host::{
+    syscall::handler::{SyscallContext, SyscallHandler},
+    syscall_types::{SyscallError, SyscallResult},
+};
+
+impl SyscallHandler {
+    #[log_syscall(/* rv */ libc::c_int, /* clockid */ libc::c_int, /* flags */ libc::c_int)]
+    pub fn timerfd_create(
+        ctx: &mut SyscallContext,
+        clockid: libc::c_int,
+        flags: libc::c_int,
+    ) -> SyscallResult {
+        // Continue only if we support the clockid.
+        check_clockid(clockid)?;
+
+        let Some(flags) = TimerFlags::from_bits(flags) else {
+            log::warn!("Invalid timerfd_create flags: {}", flags);
+            return Err(Errno::EINVAL.into());
+        };
+
+        let mut file_flags = FileStatus::empty();
+        let mut desc_flags = DescriptorFlags::empty();
+
+        if flags.contains(TimerFlags::TFD_NONBLOCK) {
+            file_flags.insert(FileStatus::NONBLOCK);
+        }
+
+        if flags.contains(TimerFlags::TFD_CLOEXEC) {
+            desc_flags.insert(DescriptorFlags::CLOEXEC);
+        }
+
+        let file = TimerFd::new(file_flags);
+        let mut desc = Descriptor::new(CompatFile::New(OpenFile::new(File::TimerFd(file))));
+        desc.set_flags(desc_flags);
+
+        let fd = ctx
+            .objs
+            .process
+            .descriptor_table_borrow_mut()
+            .register_descriptor(desc)
+            .or(Err(Errno::ENFILE))?;
+
+        log::trace!("timerfd_create() returning fd {}", fd);
+
+        Ok(fd.val().into())
+    }
+
+    #[log_syscall(/* rv */ libc::c_int, /* fd */ libc::c_int, /*curr_value*/ *const libc::c_void)]
+    pub fn timerfd_gettime(
+        ctx: &mut SyscallContext,
+        fd: libc::c_int,
+        curr_value_ptr: ForeignPtr<libc::itimerspec>,
```

Comment:
Maybe add a comment that the syscall takes a `__kernel_itimerspec`, which we assume here is the same size/layout/etc as libc's `itimerspec`.